packages/webpack-extensive-lodash-replacement-plugin/index.js

```diff
@@ -0,0 +1,461 @@
+	'isInteger',
+	'isLength',
+	'isMap',
+	'isMatch',
+	'isMatchWith',
+	'isNaN',
+	'isNative',
+	'isNil',
+	'isNull',
+	'isNumber',
+	'isObject',
+	'isObjectLike',
+	'isPlainObject',
+	'isRegExp',
+	'isSafeInteger',
+	'isSet',
+	'isString',
+	'isSymbol',
+	'isTypedArray',
+	'isUndefined',
+	'isWeakMap',
+	'isWeakSet',
+	'iteratee',
+	'join',
+	'kebabCase',
+	'keyBy',
+	'keys',
+	'keysIn',
+	'last',
+	'lastIndexOf',
+	'lodash',
+	'lowerCase',
+	'lowerFirst',
+	'lt',
+	'lte',
+	'map',
+	'mapKeys',
+	'mapValues',
+	'matches',
+	'matchesProperty',
+	'max',
+	'maxBy',
+	'mean',
+	'meanBy',
+	'memoize',
+	'merge',
+	'mergeWith',
+	'method',
+	'methodOf',
+	'min',
+	'minBy',
+	'mixin',
+	'multiply',
+	'negate',
+	'next',
+	'noop',
+	'now',
+	'nth',
+	'nthArg',
+	'omit',
+	'omitBy',
+	'once',
+	'orderBy',
+	'over',
+	'overArgs',
+	'overEvery',
+	'overSome',
+	'pad',
+	'padEnd',
+	'padStart',
+	'parseInt',
+	'partial',
+	'partialRight',
+	'partition',
+	'pick',
+	'pickBy',
+	'plant',
+	'property',
+	'propertyOf',
+	'pull',
+	'pullAll',
+	'pullAllBy',
+	'pullAllWith',
+	'pullAt',
+	'random',
+	'range',
+	'rangeRight',
+	'rearg',
+	'reduce',
+	'reduceRight',
+	'reject',
+	'remove',
+	'repeat',
+	'replace',
+	'rest',
+	'result',
+	'reverse',
+	'round',
+	'sample',
+	'sampleSize',
+	'set',
+	'setWith',
+	'shuffle',
+	'size',
+	'slice',
+	'snakeCase',
+	'some',
+	'sortBy',
+	'sortedIndex',
+	'sortedIndexBy',
+	'sortedIndexOf',
+	'sortedLastIndex',
+	'sortedLastIndexBy',
+	'sortedLastIndexOf',
+	'sortedUniq',
+	'sortedUniqBy',
+	'split',
+	'spread',
+	'startCase',
+	'startsWith',
+	'stubArray',
+	'stubFalse',
+	'stubObject',
+	'stubString',
+	'stubTrue',
+	'subtract',
+	'sum',
+	'sumBy',
+	'tail',
+	'take',
+	'takeRight',
+	'takeRightWhile',
+	'takeWhile',
+	'tap',
+	'template',
+	'templateSettings',
+	'throttle',
+	'thru',
+	'times',
+	'toArray',
+	'toFinite',
+	'toInteger',
+	'toIterator',
+	'toJSON',
+	'toLength',
+	'toLower',
+	'toNumber',
+	'toPairs',
+	'toPairsIn',
+	'toPath',
+	'toPlainObject',
+	'toSafeInteger',
+	'toString',
+	'toUpper',
+	'transform',
+	'trim',
+	'trimEnd',
+	'trimStart',
+	'truncate',
+	'unary',
+	'unescape',
+	'union',
+	'unionBy',
+	'unionWith',
+	'uniq',
+	'uniqBy',
+	'uniqWith',
+	'uniqueId',
+	'unset',
+	'unzip',
+	'unzipWith',
+	'update',
+	'updateWith',
+	'upperCase',
+	'upperFirst',
+	'value',
+	'valueOf',
+	'values',
+	'valuesIn',
+	'without',
+	'words',
+	'wrap',
+	'wrapperAt',
+	'wrapperChain',
+	'wrapperCommit',
+	'wrapperLodash',
+	'wrapperNext',
+	'wrapperPlant',
+	'wrapperReverse',
+	'wrapperToIterator',
+	'wrapperValue',
+	'xor',
+	'xorBy',
+	'xorWith',
+	'zip',
+	'zipObject',
+	'zipObjectDeep',
+	'zipWith',
+];
+
+function throwError( message, error = null ) {
+	throw new Error(
+		`[ExtensiveLodashReplacementPlugin] ${ message }${ error ? ' Error: ' : '' }${ error }`
+	);
+}
+
+class ExtensiveLodashReplacementPlugin {
+	constructor( baseFile = './package.json' ) {
+		this.baseFile = path.resolve( baseFile );
```

Comment:
In a future patch, the resolution should be done by the webpack itself. It has the [enhanced-resolve library](https://github.com/webpack/enhanced-resolve) that implements the Node resolution algorithm that can be customized by a variety of hooks.

The most precise wording of the question we want to answer here is: if I resolve `lodash` inside the app's entrypoint module, what module and what package version does it resolve to?

Writing a code to read and parse the `package.json` should also be redundant in an ideal webpack world: the data are available in the resolution's `result`'s `descriptionFileData` field.

Reply:
Oh, thanks, @jsnajdr ! I didn't know about that library. I'll definitely work on a follow-up PR switching to that 👍